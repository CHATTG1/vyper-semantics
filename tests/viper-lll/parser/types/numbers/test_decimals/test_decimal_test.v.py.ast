%pgm(, ,
  %fdecl(%@public, foo, , %num,
    %return(%floor(%fixed10(999, 1))))
  %fdecl(%@public, fop, , %num,
    %return(%floor(%binop(+, %fixed10(333, 1), %fixed10(666, 1)))))
  %fdecl(%@public, foq, , %num,
    %return(%floor(%binop(-, %fixed10(13321, 10), %fixed10(3331, 10)))))
  %fdecl(%@public, bar, , %num,
    %return(%floor(%binop(*, %fixed10(27, 1), %fixed10(37, 1)))))
  %fdecl(%@public, baz, , %num,
    %assign(%var(x), %fixed10(27, 1))
    %return(%floor(%binop(*, %var(x), %fixed10(37, 1)))))
  %fdecl(%@public, baffle, , %num,
    %return(%floor(%binop(*, %fixed10(27, 1), 37))))
  %fdecl(%@public, mok, , %num,
    %return(%floor(%binop(/, %binop(/, %binop(/, %fixed10(999999, 1), %fixed10(7, 1)), %fixed10(11, 1)), %fixed10(13, 1)))))
  %fdecl(%@public, mol, , %num,
    %return(%floor(%binop(/, %fixed10(4995, 10), %fixed10(5, 10)))))
  %fdecl(%@public, mom, , %num,
    %return(%floor(%binop(/, %fixed10(14985, 10), %fixed10(15, 10)))))
  %fdecl(%@public, mon, , %num,
    %return(%floor(%binop(/, %fixed10(2997, 1), 3))))
  %fdecl(%@public, moo, , %num,
    %return(%floor(%binop(/, 2997, %fixed10(3, 1)))))
  %fdecl(%@public, foom, , %num,
    %return(%floor(%binop(%, %fixed10(1999, 1), %fixed10(1000, 1)))))
  %fdecl(%@public, foon, , %num,
    %return(%floor(%binop(%, %fixed10(1999, 1), 1000))))
  %fdecl(%@public, foop, , %num,
    %return(%floor(%binop(%, 1999, %fixed10(1000, 1)))))
)
