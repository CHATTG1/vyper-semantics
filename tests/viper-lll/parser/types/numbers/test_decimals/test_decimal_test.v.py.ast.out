$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3264763256 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( 9990000000000 , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 67103786 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $add ( 3330000000000 , 6660000000000 ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1733984807 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $sub ( 13321000000000 , 3331000000000 ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 4273672062 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $with ( $r , 370000000000 , $with ( $l , 270000000000 , $with ( $ans , $mul ( $l , $r ) , $%seq ( $assert ( $or ( $eq ( $sdiv ( $ans , $l ) , $r ) , $not ( $l ) ) ) , $sdiv ( $ans , 10000000000 ) ) ) ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 2811326380 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $mstore ( 320 , 270000000000 ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $with ( $r , 370000000000 , $with ( $l , $mload ( 320 ) , $with ( $ans , $mul ( $l , $r ) , $%seq ( $assert ( $or ( $eq ( $sdiv ( $ans , $l ) , $r ) , $not ( $l ) ) ) , $sdiv ( $ans , 10000000000 ) ) ) ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 207862020 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $with ( $r , 37 , $with ( $l , 270000000000 , $with ( $ans , $mul ( $l , $r ) , $%seq ( $assert ( $or ( $eq ( $sdiv ( $ans , $l ) , $r ) , $not ( $l ) ) ) , $ans ) ) ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 793503074 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $with ( $l , $clamp ( $mload ( 160 ) , $with ( $l , $clamp ( $mload ( 160 ) , $with ( $l , 9999990000000000 , $with ( $r , $clamp_nonzero ( 70000000000 ) , $sdiv ( $mul ( $l , 10000000000 ) , $r ) ) ) , $mload ( 128 ) ) , $with ( $r , $clamp_nonzero ( 110000000000 ) , $sdiv ( $mul ( $l , 10000000000 ) , $r ) ) ) , $mload ( 128 ) ) , $with ( $r , $clamp_nonzero ( 130000000000 ) , $sdiv ( $mul ( $l , 10000000000 ) , $r ) ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 3822704924 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $with ( $l , 4995000000000 , $with ( $r , $clamp_nonzero ( 5000000000 ) , $sdiv ( $mul ( $l , 10000000000 ) , $r ) ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1039648877 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $with ( $l , 14985000000000 , $with ( $r , $clamp_nonzero ( 15000000000 ) , $sdiv ( $mul ( $l , 10000000000 ) , $r ) ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 3888573150 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $sdiv ( 29970000000000 , $clamp_nonzero ( 3 ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1674418635 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $sdiv ( $mul ( 2997 , 100000000000000000000 ) , $clamp_nonzero ( 30000000000 ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 553684216 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $smod ( 19990000000000 , $clamp_nonzero ( 10000000000000 ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 262072420 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $smod ( 19990000000000 , $mul ( $clamp_nonzero ( 1000 ) , 10000000000 ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 2434275585 ) , $%seq ( $pass , $assert ( $iszero ( $callvalue ) ) , $%seq ( $mstore ( 0 , $sdiv ( $clamp ( $mload ( 160 ) , $smod ( $mul ( 1999 , 10000000000 ) , $clamp_nonzero ( 10000000000000 ) ) , $mload ( 128 ) ) , 10000000000 ) ) , $return ( 0 , 32 ) ) , $stop ) ) ) , 0 ) ) )
